lib/improver/wind_calculations/wind_downscaling.py

```diff
@@ -654,12 +654,10 @@ def __init__(self, a_over_s_cube, sigma_cube, pporo_cube,
                 2D - vegetative roughness length in m. If not given, do not do
                 any RC
         """
-        enforce_float32_precision([a_over_s_cube,
-                                   sigma_cube,
-                                   pporo_cube,
-                                   modoro_cube,
-                                   z0_cube,
-                                   height_levels_cube])
+        for cube in [a_over_s_cube, sigma_cube, pporo_cube, modoro_cube,
+                     z0_cube, height_levels_cube]:
+            if cube is not None:
+                check_cube_not_float64(cube, fix=True)
 
         # Standard Python 'float' type is either single or double depending on
         # system and there is no reliable method of finding which from the
```

Comment:
Comment needs updating

Reply:
This comment refers to the line below (updating a single float value), which I haven't changed.

Reply:
Agreed